### ISSUE TITLE
Failed to load the librdkafka native library on Amazon Linux Instance. (Confluent-Kafka version 0.11.3)

### ISSUE THREAD

**Issue by InKolev** [confluentinc/confluent-kafka-dotnet#452]
Description
===========
Hi, I am trying to run a **.net core 2.0** application on an **Amazon Linux** instance,
that has **Confluent.Kafka** nugget version **0.11.3** as a dependency.
It's running perfectly fine on my Windows 10 OS locally.
But when I do the following:

1. Publish the app **targeting linux-x64 runtime**:
```
dotnet publish .\MicroServices.SQS.To.Kafka.Bridge.csproj -c Release -r linux-x64
```

2. Copy-paste the resulting publish folder on the Amazon Linux instance.

3. Run the application from the "publish" folder:
```
dotnet MicroServices.SQS.To.Kafka.Bridge.dll
```

**The result is the following exception:**
```
Unhandled Exception: System.DllNotFoundException: Failed to load the librdkafka native library.
   at Confluent.Kafka.Impl.LibRdKafka.Initialize(String userSpecifiedPath)
   at Confluent.Kafka.Producer..ctor(IEnumerable`1 config, Boolean manualPoll, Boolean disableDeliveryReports)
   at Confluent.Kafka.Producer`2..ctor(IEnumerable`1 config, ISerializer`1 keySerializer, ISerializer`1 valueSerializer, Boolean manualPoll, Boolean disableDeliveryReports)
   at MicroServices.SQS.To.Kafka.Bridge.Configuration.SqsToKafkaBridgeConfigurator.RegisterKafkaProducer(ContainerBuilder containerBuilder, KafkaProducerSettings settings) in E:\Catalytic\microservices.sqs.to.kafka.bridge\src\MicroServices.SQS.To.Kafka.Bridge\Configuration\SqsToKafkaBridgeConfigurator.cs:line 57
   at MicroServices.SQS.To.Kafka.Bridge.Configuration.SqsToKafkaBridgeConfigurator.Configure(ContainerBuilder containerBuilder, JObject serviceConfig) in E:\Catalytic\microservices.sqs.to.kafka.bridge\src\MicroServices.SQS.To.Kafka.Bridge\Configuration\SqsToKafkaBridgeConfigurator.cs:line 28
   at MicroServices.Infrastructure.MicroServiceInitializer.ConfigureDependencyInjectionContainer()
   at MicroServices.Infrastructure.MicroServiceInitializer.Initialize()
   at MicroServices.SQS.To.Kafka.Bridge.Startup.Main(String[] args) in E:\Catalytic\microservices.sqs.to.kafka.bridge\src\MicroServices.SQS.To.Kafka.Bridge\Startup.cs:line 10
Aborted
```

**Do you guys think it might be related to the "published" application folder structure?
Because it seems like the loaders have troubles finding the librdkafka dll's.**

The "published"  application folder structure is the following:
![image](https://user-images.githubusercontent.com/6628205/37114004-bf607c94-224f-11e8-800f-6eed1deb13a4.png)

![image](https://user-images.githubusercontent.com/6628205/37114037-dba92e14-224f-11e8-843b-3368a518efcd.png)

**CPU Architecture of the Amazon Linux instance is:**
Architecture: x86_64
CPU op-mode(s): 32-bit, 64-bit

I will really appreciate if someone has any idea what might be the cause of this issue, because it's a huge blocker for us in terms of release dates. If you need more information regarding the issue, just ask in the comments and I will do my best to provide it.

Thanks in advance.

**Comment by ah-** [comment]
It's been a while since I've touched this but I'd look into two things:
- Does dotnet pick a librdkafka.so for Amazon Linux? I think it has some concept of distros, does that detect Amazon Linux correctly? Since you did a publish and it included a librdkafka.so I guess it does.
- Does Amazon Linux have all dependencies installed? To check log onto an instance and run `ldd librdkafka.so`.

**Comment by InKolev** [comment]
@ah- 
I tried targeting CentOS also, but it didn't work.

I actually just **solved the problem** by **downloading librdkafka source** from github and **compiling it on the Amazon Linux instance itself**.

When I replaced the dll that was generated from the dotnet publish command with the one that I manually created, the application started successfully.

So yeah, I guess it's because **Amazon Linux is not an officially supported distro** we cannot expect everything to work seamlessly.

Thanks for the guidance, bro. I am closing the issue now. :)
